### PR TITLE
# 156882155  Implement login with jwt

### PR DESCRIPTION
**Pull request by emmabaye**
#### What does this PR do?
Implement login API with jwt.
#### Description of Task to be completed?
N/A
#### How should this be manually tested?
Make a post request to `/api/auth/login` with body `{email, password}`. Successful response should include jwt.
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#156882155 
#### Questions:
N/A

### CODE REVIEW

**Review by Orlayhemmy — COMMENTED** [review]
Good job for fixing the eslint errors

File: config/middlewares/authorization.js [hunk 1]
```diff
@@ -1,33 +1,23 @@
-/**
- * Generic require login routing middleware
- */
-exports.requiresLogin = function(req, res, next) {
-    if (!req.isAuthenticated()) {
-        return res.send(401, 'User is not authorized');
-    }
-    next();
+/* eslint-disable func-names */
+/* eslint-disable no-underscore-dangle */
```
Comment (chuzksy-codeactive) [inline config/middlewares/authorization.js:2]: Good use of eslint inline configuration to disable some rules

File: app/controllers/users.js [hunk 2]
```diff
@@ -1,99 +1,104 @@
...
+exports.signup = function (req, res) {
   if (!req.user) {
     res.redirect('/#!/signup');
   } else {
     res.redirect('/#!/app');
   }
 };
 
+exports.loginSuccess = (req, res) => {
+  console.log('Right details');
```
Comment (andela-moe) [inline app/controllers/users.js:36]: Do we really need this `console.log('Right details');` here? If this was used during development then you can take this.

File: app/controllers/users.js [hunk 3]
```diff
@@ -1,99 +1,104 @@
...
+    data: {
+      token,
+      authenticated: true,
+      user: req.user
+    }
+  });
+};
+
+exports.loginFailure = (err, req, res) => {
+  console.log('Wrong Details');
```
Comment (andela-moe) [inline app/controllers/users.js:53]: Do we really need this `console.log('Wrong details'); `here? If this was used during development then you can take this.

File: app/controllers/users.js [hunk 4]
```diff
@@ -107,46 +112,31 @@ exports.create = function(req, res) {
...
-            duplicate = true;
+          let duplicate = false;
+          for (let i = 0; i < user.donations.length; i++) { //eslint-disable-line
+            if (user.donations[i].crowdrise_donation_id
+              === req.body.crowdrise_donation_id) {
+              duplicate = true;
+            }
+          }
+          if (!duplicate) {
+            console.log('Validated donation');
```
Comment (andela-moe) [inline app/controllers/users.js:134]: Do we really need this `console.log('Validated donation');` here? If this was used during development then you can take this.

**Review by andela-moe — COMMENTED** [review]
You have `console.log()` in certain parts of the files you have in this PR. Do you need them or were they used specifically for development purposes?

You seemed to have commented/deleted out the tests for the User model. Please confirm my assumptions.

Good enough these files have no comments from HoundCI. It is obvious that you have disabled certain files and some lines also. I believe there are better ways of having to deal with the issues flagged by eslint.

Please review comments and resolve

File: app/controllers/users.js (hunk 4 above)
Comment (emmabaye) [reply app/controllers/users.js:134]: The console.log() statements was used for development and I deleted the user test. I will fix that right away.

**Pull request closed without merging**